internal/service/user.go

```diff
@@ -24,56 +25,148 @@ func NewUserService(repo UserRepo, l logger.ZeroLog) UserService {
 }
 
 func (s UserService) Create(user entity.User) error {
-	// TODO: Implement me!
-	return nil
+	err := s.validateUser(user)
+	if err != nil {
+		return err
+	}
+	return s.repo.Create(user)
 }
 
 func (s UserService) Get(id uint64) (entity.User, error) {
-	// TODO: Implement me!
-	return entity.User{}, nil
+	user, err := s.repo.Read(id)
+	if err != nil {
+		return entity.User{}, err
+	}
+	return user, nil
 }
 
 func (s UserService) GetAll() ([]entity.User, error) {
-	// TODO: Implement me!
-	return nil, nil
+	return s.repo.ReadAll()
 }
 
 func (s UserService) Find(filter, value string) ([]entity.User, error) {
-	// TODO: Implement me!
-	return nil, nil
+	err := validateFilter(filter)
+	if err != nil {
+		return []entity.User{}, err
+	}
+
+	users, err := s.repo.ReadAll()
+	if err != nil {
+		return []entity.User{}, err
+	}
+
+	filteredUsers := []entity.User{}
+	for _, user := range users {
+		switch {
+		case filter == "FirstName":
+			if user.FirstName == value {
+				filteredUsers = append(filteredUsers, user)
+			}
+		case filter == "LastName":
+			if user.LastName == value {
+				filteredUsers = append(filteredUsers, user)
+			}
+		case filter == "Email":
+			if user.Email == value {
+				filteredUsers = append(filteredUsers, user)
+			}
+		case filter == "Username":
+			if user.Username == value {
+				filteredUsers = append(filteredUsers, user)
+			}
+		}
+
+	}
+	return filteredUsers, nil
 }
 
 func (s UserService) Update(user entity.User) error {
-	// TODO: Implement me!
-	return nil
+	err := s.validateUser(user)
+	if err != nil {
+		return err
+	}
+	return s.repo.Update(user)
 }
 
 func (s UserService) Delete(id uint64) error {
-	// TODO: Implement me!
-	return nil
+	return s.repo.Delete(id)
 }
 
 func (s UserService) Activate(id uint64) error {
-	// TODO: Implement me!
-	return nil
+	user, err := s.repo.Read(id)
+	if err != nil {
+		return err
+	}
+	user.Active = true
+	return s.repo.Update(user)
 }
 
 func (s UserService) ChangeEmail(id uint64, email string) error {
-	// TODO: Implement me!
-	return nil
+	user, err := s.repo.Read(id)
+	if err != nil {
+		return err
+	}
+	user.Email = email
+	return s.repo.Update(user)
 }
 
 func (s UserService) ChangePasswd(id uint64, passwd string) error {
-	// TODO: Implement me!
-	return nil
+	err := validatePassword(passwd)
+	if err != nil {
+		return err
+	}
+	user, err := s.repo.Read(id)
+	if err != nil {
+		return err
+	}
+	user.Passwd = passwd
+	return s.repo.Update(user)
 }
 
 func (s UserService) IsActive(id uint64) (bool, error) {
-	// TODO: Implement me!
-	return false, nil
+	user, err := s.repo.Read(id)
+	if err != nil {
+		return false, err
+	}
+	return user.Active, nil
 }
 
 func (s UserService) ValidateLogin(username string, password string) (entity.User, error) {
-	// TODO: Implement me!
-	return entity.User{}, nil
+	users, err := s.Find("Username", username)
+	if err != nil {
+		return entity.User{}, err
+	}
+	if users[0].Passwd == password {
```

Comment:
We can't validate a plain password vs a hashed password. I suspect we should implement a ValidateLogin function in the repository implementation or add more logic in the service function